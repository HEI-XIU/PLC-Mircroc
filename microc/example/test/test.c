  int a[5];
void main(int n) {
  
    // a[1]=1;
    // a[6]=10;
    // int i =(int) 1.4;
    // float f = 1.1;
    // char c = 'c';
    // string s = "let me see";
    // printf("%d",i);
    // printf("%f",f);
    // printf("%c",c);
    // printf("%s",s);
    // printf("%d",n);

    // int i=1;
    // i+=2;
    // printf("%d",i);
    // i*=2;
    // printf("%d",i);
    // printf("%d",a[1]);
    // printf("%d",a[6]);

    // printf("%d",(Int)12.5 );
    
  int i = 0;
  print i;
    // char i = 1;
    // printf("%d",(Int) '9' );
    // printf("%c",(Char)1 );
    // printf("%s",(String)12 );

    // printf("%f",(Float)13);

}