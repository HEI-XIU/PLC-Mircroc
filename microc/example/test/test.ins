INCSP 5
GETSP
OFFSET 4
SUB
LDARGS 1
CALL (1, "L1")
STOP
FLabel (1, "L1")
INCSP 1
GETBP
OFFSET 1
ADD
CSTI 0
STI
INCSP -1
GETBP
OFFSET 1
ADD
LDI
PRINTI
INCSP -1
INCSP -1
RET 0