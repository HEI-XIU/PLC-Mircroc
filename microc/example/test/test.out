15 5 14 0 4 2 24 19 1 11 25 15 1 13 0 1 1 0 0 12 15 -1 13 0 1 1 11 22 15 -1 15 -1 21 0